applications/ParticleMechanicsApplication/custom_utilities/mpm_explicit_utilities.cpp

```diff
@@ -237,15 +237,15 @@ namespace Kratos
     /***********************************************************************************/
 
     void MPMExplicitUtilities::CalculateMUSLGridVelocity(
-        Element& rElement,
-        const Vector& rN)
+        Element& rElement)
     {
         KRATOS_TRY
 
         GeometryType& rGeom = rElement.GetGeometry();
         const SizeType dimension = rGeom.WorkingSpaceDimension();
         const SizeType number_of_nodes = rGeom.PointsNumber();
         const ProcessInfo& rProcessInfo = ProcessInfo();
```

Comment:
creating on the fly?
ouch :sweat_smile:

Reply:
Nay sir, it is the magic of the quadrature point!

Reply:
yikes!
I would consider this a blocker, but up to you

Reply:
The shape functions are accessed (not created) from the element geom now due to the quad point change. Check it out [here](https://github.com/KratosMultiphysics/Kratos/blob/d1e898540c321a636b6f32ac53da3d61601206c4/applications/ParticleMechanicsApplication/custom_utilities/mpm_explicit_utilities.cpp#L248). That's why we can remove the shape function arg of the function.

Reply:
I mean why not calling the function `CalculateMUSLGridVelocity` with an instance of `ProcessInfo`?

Reply:
Ah ok, yep that makes sense and is much safer. I'll do that now, thanks!